Code/Framework/AzFramework/Platform/Linux/AzFramework/Windowing/NativeWindow_Linux_xcb.cpp

```diff
@@ -0,0 +1,244 @@
+        }
+        LinuxXcbEventHandlerBus::Handler::BusDisconnect();
+    }    
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    NativeWindowHandle NativeWindowImpl_Linux_xcb::GetWindowHandle() const
+    {
+        return reinterpret_cast<NativeWindowHandle>(m_xcbWindow);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::SetWindowTitle(const AZStd::string& title)
+    {
+        xcb_void_cookie_t xcb_check_result;
+        xcb_check_result = xcb_change_property(m_xcbConnection,
+                                               XCB_PROP_MODE_REPLACE,
+                                               m_xcbWindow,
+                                               XCB_ATOM_WM_NAME,
+                                               XCB_ATOM_STRING,
+                                               8,
+                                               static_cast<uint32_t>(title.size()),
+                                               title.c_str());
+        AZ_Assert(ValidateXcbResult(xcb_check_result), "Failed to set window title.");
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::ResizeClientArea(WindowSize clientAreaSize)
+    {
+        const static uint32_t values[] = { clientAreaSize.m_width, clientAreaSize.m_height };
+
+        xcb_configure_window (m_xcbConnection, m_xcbWindow, XCB_CONFIG_WINDOW_WIDTH | XCB_CONFIG_WINDOW_HEIGHT, values);
+        
+        m_width = clientAreaSize.m_width;
+        m_height = clientAreaSize.m_height;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    uint32_t NativeWindowImpl_Linux_xcb::GetDisplayRefreshRate() const
+    {
+        //Using 60 for now until proper support is added
+        return 60;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    bool NativeWindowImpl_Linux_xcb::ValidateXcbResult(xcb_void_cookie_t cookie)
+    {
+        bool result = true;
+        if (xcb_generic_error_t* error = xcb_request_check(m_xcbConnection, cookie))
+        {
+            AZ_TracePrintf("Error","Error code %d", error->error_code);
+            result = false;
+        }
+        return result;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::HandleXcbEvent(xcb_generic_event_t* event)
+    {
+        switch (event->response_type & 0x7f)
+        {
+            case XCB_CONFIGURE_NOTIFY:
+            {
+                xcb_configure_notify_event_t *cne = reinterpret_cast<xcb_configure_notify_event_t *>(event);
+                if ((cne->width != m_width) ||
+                    (cne->height != m_height))
```

Comment:
This is already being checked in NativeWindowImpl_Linux_xcb::WindowSizeChanged so maybe not necessary here?

Reply:
Removed check